src/App.js

```diff
@@ -1,26 +1,30 @@
 import React from 'react';
-import logo from './logo.svg';
+import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
+import Sidebar from './Layout/Sidebar';
+import UserSettingsProvider from './context/userSettingsContext';
+import ChatsProvider from './context/chatsContext';
+import * as Pages from './Pages';
 import './App.css';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+const App = () => (
+    <div className="app">
+        <Router>
+            <UserSettingsProvider>
+                <Sidebar />
+                <div className="main">
+                    <Switch>
+                        <Route exact path="/" component={Pages.Home} />
+                        <Route path="/settings" component={Pages.UserSettings} />
+                        <Route path="/home" component={Pages.Home} />
+                        <ChatsProvider>
```

Comment:
you should move the Provider to inner or outer scope (the route switch stays clean that way)